lib/proxyquire.js

```diff
@@ -175,18 +215,8 @@ Proxyquire.prototype._withoutCache = function (module, stubs, path, func) {
     if (this._preserveCache) {
       restoreCache()
     } else {
-      var id = Module._resolveFilename(path, module)
-      var stubIds = Object.keys(stubs).map(function (stubPath) {
-        try {
-          return resolve.sync(stubPath, {
-            basedir: dirname(id),
-            extensions: Object.keys(require.extensions),
-            paths: Module.globalPaths
-          })
-        } catch (_) {}
-      })
-      var ids = [id].concat(stubIds.filter(Boolean))
-
+      var stubIds = Object.keys(stubs)
```

Comment:
You can merge this line with the next now (`Object.keys(stubs).filter(Boolean)`)